File: plox/tests/test_stack.py
import sys

sys.path.append("../src/")
from stack import Stack


def test_stack():
    st = Stack()

    st.push(5)
    st.push(7)
    st.push(9)
    assert st.get(0) == 9
    assert st.pop() == 9
    assert st.length == 2

    st.push(11)

    assert st.get(0) == 11 
    assert st.pop() == 11
    assert st.get(1) == 5
    assert st.pop() == 7
    assert st.peek() == 5
    assert st.pop() == 5
    assert st.pop() is None

    st.push(69)
    assert st.peek() == 69
    assert st.length == 1

File: cpplox/test/TestStack.cpp

#include "../src/stack.h"
#include <gtest/gtest.h>

TEST(TestStack, TestStack) {
  Stack *stack = new Stack();
  stack->init();
  stack->length = 0;

  Value *v1 = new Value(5.0);
  stack->push(v1);

  Value *v2 = new Value(7.0);
  stack->push(v2);

  Value *v3 = new Value(9.0);
  stack->push(v3);
  EXPECT_EQ(stack->length, 3);

  EXPECT_EQ(stack->pop(), 9.0);
  EXPECT_EQ(stack->length, 2);

  Value *v4 = new Value(11.0);
  stack->push(v4);
  EXPECT_EQ(stack->pop(), 11.0);

  EXPECT_EQ(stack->pop(), 7.0);
  EXPECT_EQ(*stack->peek(), 5.0);
  EXPECT_EQ(stack->pop(), 5.0);
  std::cout << stack->length << "\n";
  EXPECT_EQ(stack->pop(), NULL);

  Value *v5 = new Value(69.0);
  stack->push(v5);
  EXPECT_EQ(*stack->peek(), 69.0);
  EXPECT_EQ(stack->length, 1);
}
